File: programmers/etc/level2/가장_큰_수.py
def solution(numbers):
    numbers.sort(reverse=True, key = lambda x: str(x)*3) # numbers 원소 3자리수 이하 이용
    numbers = "".join(str(num) for num in numbers)
    return "0" if numbers[0] == "0" else numbers


numbers = [6, 10, 2]
print(solution(numbers)) # "6210"

numbers = [3, 30, 34, 5, 9]
print(solution(numbers)) # "9534330"

File: Java/programmers/etc/level2/가장큰수.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 가장큰수 {
    public String solution(int[] numbers) {
        String[] nums = new String[numbers.length];

        for (int i=0; i<numbers.length; i++) {
            nums[i] = String.valueOf(numbers[i]);
        }

        Arrays.sort(nums, new Comparator<String>() {
            public int compare(String o1, String o2) {
                return (o2+o1).compareTo(o1+o2);
            }
        });

        String answer = "";
        for (String num : nums) {
            answer += num;
        }

        return answer.charAt(0) == '0' ? "0" : answer;
    }
}
